File: binary_search/367. Valid Perfect Square/367.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if (num == 1): return True
        
        l = 0
        r = num
        while r - l > 1:
            
            m = (r + l) // 2
            if (m ** 2 == num): return True
            if (m ** 2 > num):
                r = m
            else:
                l = m
        return False

File: binary_search/367. Valid Perfect Square/367.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isPerfectSquare = function(num) {
    if (num === 1) return true
    let l = 0
    let r = num
    while ( r - l > 1) {
        let m = Math.floor((r + l) / 2)

        if (m * m == num) return true
        if (m * m > num) r = m
        else l = m
    }

    return false
};